app/models/concerns/client_search.rb

```diff
@@ -89,5 +89,11 @@ def self.metaphone_search(where, field, text)
 
       where
     end
+
+    def self.with_custom_name_search(where, field, text, association: :names)
+      return where unless reflect_on_association(association)&.klass == Hmis::Hud::CustomClientName
+
+      where.or(Hmis::Hud::CustomClientName.arel_table[field].lower.matches("#{text&.downcase}%"))
+    end
```

Comment:
We aren't super careful about this, but we try to prevent bleed from the drivers into the main app.  Could we change this to:
```suggestion
      return where unless RailsDrivers.loaded.include?(:hmis)
      return where unless reflect_on_association(association)&.klass == Hmis::Hud::CustomClientName

      where.or(Hmis::Hud::CustomClientName.arel_table[field].lower.matches("#{text&.downcase}%"))
    end
```

I was thinking that you should be able to include the left_outer_joins here as well, but unless we want to write the join in arel, which is nasty and forces you to also handle the acts as paranoid stuff manually, I don't think it's worth it.

Reply:
@eanders Yeah, something like that sounds good. The other option would be to pass all the custom name stuff as arguments to `text_searcher` so that it doesn't need to be aware of `Hmis::Hud::CustomClientName` at all. I think that option is better since we should be able to keep the driver-specific stuff totally out of this code if we do that. I'll give that a shot and see if I can make it work.

And yeah, doing the join in arel seemed pretty difficult to do correctly so I also came to the conclusion it wasn't worth it.